include/mbedtls/x509_csr.h

```diff
@@ -205,6 +205,12 @@ void mbedtls_x509write_csr_set_md_alg( mbedtls_x509write_csr *ctx, mbedtls_md_ty
  * \param key_usage key usage flags to set
  *
  * \return          0 if successful, or MBEDTLS_ERR_X509_ALLOC_FAILED
+ *
+ * \note            The decipherOnly flag from the Key Usage extension is
+ *                  represented by bit 8 (i.e. 0x8000), which cannot typically
+ *                  be reprensented in an unsigned char. Therefore, the flag
+ *                  decipherOnly (i.e. MBEDTLS_X509_KU_DECIPHER_ONLY) cannot be
+ *                  set using this function.
```

Comment:
@hanno-arm: This does not actually fix https://github.com/ARMmbed/mbedtls/issues/1605, but it does clarify the documentation. The real fix requires deprecating this API and creating a new one with the correct input type, but perhaps this is out of the scope of this PR.

Reply:
@andresag01 I agree that this should be treated differently, but I think it's a good opportunity to at least point out the present limitation in the documentation.

Reply:
No problem. I am not disagreeing with the documentation change, I think its good to include it. I just added the comment for the benefit of other reviewers so that they dont have to look at the whole conversation to figure our reasoning.